server/events/vcs/github_client.go

```diff
@@ -72,14 +73,15 @@ type GithubAppTemporarySecrets struct {
 }
 
 // NewGithubClient returns a valid GitHub client.
-func NewGithubClient(hostname string, credentials GithubCredentials, logger logging.SimpleLogging, mergeabilityChecker MergeabilityChecker) (*GithubClient, error) {
+func NewGithubClient(hostname string, credentials GithubCredentials, logger logging.SimpleLogging, mergeabilityChecker MergeabilityChecker, useGithubChecks bool) (*GithubClient, error) {
```

Comment:
Why does the client need to know about this flag?  Seems like you could just create a proxy `StatusUpdater` that handles the delegation.  This method is a mess already, let's not add to it.